lib/BuildSystem/LaneBasedExecutionQueue.cpp

```diff
@@ -534,10 +616,98 @@ class LaneBasedExecutionQueue : public BuildExecutionQueue {
             StringRef(buf, numBytes));
       }
 
-      // Close the read end of the pipe.
+      if (readfds[0].revents & POLLIN) {
+        ssize_t numBytes = read(releasePipe[0], buf, sizeof(buf));
```

Comment:
Should we care about the case where we didn't get the full read? Also, I think it would be get to require the id be newline terminated.

Let's also make sure we document the protocol somewhere as part of this PR.